org.eclipse.xtext.xbase.testing/testdata/testdata/ShorthandNames.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package testdata;
+
+/**
+ * @author dietrich - Initial contribution and API
+ */
+public class ShorthandNames {
+
+	private int fake;
+	private int UPPER;
+	private int lower;
+	private int camelCase;
+	private int UPper;
+	private int x;
+
+	private boolean UPPERBOOL;
+	private boolean lowerbool;
+	private boolean camelCaseBool;
+	private boolean UPperBool;
+
+	public int get() {
```

Comment:
tryGetShorthand does contain logic for methods like `getuPperCase`. Can you please double-check that this is in sync with java.beans.Introspector and add a testcase?

Reply:
I believe `getuPperCase` should allow you to access the getter through `uPperCase`, but this is not the case in Xtend.

Before testing serialization we test that there are no errors in the parse result. There is an error as it thinks you're trying to access a private field instead of a getter method.
As thus, I cannot write the test as the logic is never hit.

Reply:
According to java.beans.PropertyDescriptor.getBaseName() I would think that `getuPper` cannot be aliased by `uPper`

Reply:
Isn't getBase name the other way around though? I think that method is used to generate accessors from variables: `uPper => get/setuPper`

Reply:
getBaseName is used to compute the basename, e.g. the name after the get prefix. It'll always do an toUpperCase on the first letter. So the read- and write methods cannot be found afterwards since the code will look for getUPper rather than getuPper. So the property access wouldn't work. That is my understanding.

Reply:
So the shortened name for `getuPper` should be `UPper`? That's not how it works in Xtend. In Xtend there is no shortened name for `getuPper` at all. In any case the logic will have to be adjusted

Reply:
No, actually not. In fact it's not 100% symetrical. The shorthand name for getuPper would be uPper, but later on it would fail to bind the property to the method (if we'd be an interpreted language with reflective bean properties). That's why I'D consider getuPper to be not a valid property accessor.